SignalServiceKit/src/Account/TSAccountManager.h

```diff
@@ -65,6 +65,15 @@ extern NSString *const kNSNotificationName_LocalNumberDidChange;
                       success:(void (^)())successBlock
                       failure:(void (^)(NSError *error))failureBlock;
 
+- (void)registerForManualMessageFetchingWithSuccess:(void (^)())successBlock
+                                            failure:(void (^)(NSError *error))failureBlock;
+
+// Called once registration is complete - meaning the following have succeeded:
+// - obtained signal server credentials
+// - uploaded pre-keys
+// - uploaded push tokens
```

Comment:
Don't we do this sometimes even if upload of push tokens has failed?

Reply:
Well - we no longer do. 

Previously we'd show an error, but then if you restarted Signal you'd be in a state where you could send messages, but no one could message you.